src/protocol/ipa/mod.rs

```diff
@@ -963,79 +963,96 @@ pub mod tests {
-        let records_sent = snapshot.get_counter(RECORDS_SENT);
-        assert!(records_sent <= RECORDS_SENT_SEMI_HONEST_BASELINE);
-        if records_sent < RECORDS_SENT_SEMI_HONEST_BASELINE {
-            tracing::warn!("Baseline for semi-honest IPA has improved! Expected {RECORDS_SENT_SEMI_HONEST_BASELINE}, got {records_sent}.\
-                            Strongly consider adjusting the baseline, so the gains won't be accidentally offset by a regression.");
-        }
+        for per_user_cap in [1, 3] {
+            let world = TestWorld::new_with(*TestWorldConfig::default().enable_metrics()).await;
 
-        let world = TestWorld::new_with(*TestWorldConfig::default().enable_metrics()).await;
+            let records: Vec<GenericReportTestInput<Fp32BitPrime, MatchKey, BreakdownKey>> = ipa_test_input!(
+                [
+                    { match_key: 12345, is_trigger_report: 0, breakdown_key: 1, trigger_value: 0 },
+                    { match_key: 12345, is_trigger_report: 0, breakdown_key: 2, trigger_value: 0 },
+                    { match_key: 68362, is_trigger_report: 0, breakdown_key: 1, trigger_value: 0 },
+                    { match_key: 12345, is_trigger_report: 1, breakdown_key: 0, trigger_value: 5 },
+                    { match_key: 68362, is_trigger_report: 1, breakdown_key: 0, trigger_value: 2 },
+                ];
+                (Fp32BitPrime, MatchKey, BreakdownKey)
+            );
 
-        let _ = world
-            .semi_honest(records, |ctx, input_rows| async move {
-                ipa_malicious::<Fp32BitPrime, MatchKey, BreakdownKey>(
-                    ctx,
-                    &input_rows,
-                    PER_USER_CAP,
-                    MAX_BREAKDOWN_KEY,
-                    NUM_MULTI_BITS,
-                )
+            let _: Vec<GenericReportTestInput<Fp32BitPrime, MatchKey, BreakdownKey>> = world
+                .semi_honest(records.clone(), |ctx, input_rows| async move {
+                    ipa::<Fp32BitPrime, MatchKey, BreakdownKey>(
+                        ctx,
+                        &input_rows,
+                        per_user_cap,
+                        MAX_BREAKDOWN_KEY,
+                        NUM_MULTI_BITS,
+                    )
+                    .await
+                    .unwrap()
+                })
                 .await
-                .unwrap()
-            })
-            .await;
+                .reconstruct();
+
+            let snapshot = world.metrics_snapshot();
+            let records_sent = snapshot.get_counter(RECORDS_SENT);
+            let semi_honest_baseline = if per_user_cap == 1 {
+                RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_1
+            } else {
+                RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_3
+            };
+            assert!(records_sent <= semi_honest_baseline,
+                "Baseline for semi-honest IPA (cap = {per_user_cap}) has DEGRADED! Expected {semi_honest_baseline}, got {records_sent}.");
+
```

Comment:
Added this nice error message for the semi-honest case. We only had it for the malicious case for some reason.

Reply:
That's my fault :/ I just added for malicious :(